### PR TITLE
apiserver: add validating admission tests

### PR DESCRIPTION
**Pull request by sttts**
Follow-up of https://github.com/kubernetes/kubernetes/pull/54484

This includes tests
- in endpoint tests,
- in generic registry,
- in patch handler,
- in admission chain.

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/admission/chain_test.go [hunk 1]
```diff
@@ -39,22 +39,23 @@ func (h *FakeHandler) Admit(a Attributes) (err error) {
...
-		Handler: NewHandler(ops...),
+		name:     name,
+		admit:    accept,
+		validate: accept,
+		Handler:  NewHandler(ops...),
 	}
 }
 
-func TestAdmit(t *testing.T) {
+func TestAdmitAndVvalidate(t *testing.T) {
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/admission/chain_test.go]: vvalidate typo

File: staging/src/k8s.io/apiserver/pkg/endpoints/apiserver_test.go [hunk 2]
```diff
@@ -2971,38 +2990,42 @@ func TestUpdate(t *testing.T) {
...
 
 func TestUpdateInvokesAdmissionControl(t *testing.T) {
-	storage := map[string]rest.Storage{}
-	simpleStorage := SimpleRESTStorage{}
-	ID := "id"
-	storage["simple"] = &simpleStorage
-	handler := handleDeny(storage)
-	server := httptest.NewServer(handler)
-	defer server.Close()
+	for _, admit := range []admission.Interface{alwaysMutatingDeny{}, alwaysValidatingDeny{}} {
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/endpoints/apiserver_test.go]: I'm not sure these are checking what we want.  We need to make sure that generic store calls admission.  This is ensuring that our mock calls it.  It's good that our mock calls it, but this is just checking the mock, not the actual code, right?

File: staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store_test.go [hunk 3]
```diff
@@ -311,8 +311,14 @@ func TestStoreCreate(t *testing.T) {
 	defaultDeleteStrategy := testRESTStrategy{scheme, names.SimpleNameGenerator, true, false, true}
 	registry.DeleteStrategy = testGracefulStrategy{defaultDeleteStrategy}
 
+	// create the object with denying admission
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store_test.go]: these are the tests I was looking for

**Comment by deads2k** [comment]
fix the typo and squash. lgtm

/approve no-issue

File: staging/src/k8s.io/apiserver/pkg/endpoints/apiserver_test.go (hunk 2 above)
Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/endpoints/apiserver_test.go]: Not on this level. The generic store has its own tests which are more e2e.

**Comment by sttts** [comment]
Fixed typo and squashed.

**Pull request merged**